### PR TITLE
Feat/backend auth setup

### PR DESCRIPTION
**Pull request by 9j3**

### CODE REVIEW

File: packages/api/src/requests.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export class LoginRequest {
+  @IsNotEmpty({ message: 'A username is required' })
+  readonly username: string;
+
+  @IsNotEmpty({ message: 'A password is required to login' })
+  readonly password: string;
+}
+
+export class RegisterRequest {
+  @IsNotEmpty({ message: 'An username is required' })
```
Comment (bariod) [inline packages/api/src/requests.ts:12]:
```suggestion
  @IsNotEmpty({ message: 'A username is required' })
```

File: packages/api/test/e2e/app/app.e2e-spec.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
 import {INestApplication} from '@nestjs/common';
 import {Test} from '@nestjs/testing';
 import * as request from 'supertest';
-import {AppModule} from "../../../src/app.module";
+import {ApplicationModule} from "../../../src/app.module";
```
Comment (bariod) [inline packages/api/test/e2e/app/app.e2e-spec.ts:4]: run the formatter on this file
```suggestion
import { ApplicationModule } from "../../../src/app.module";
```

File: packages/api/test/e2e/app/app.e2e-spec.ts [hunk 3]
```diff
@@ -17,19 +17,65 @@ describe('E2E JWT Test', () => {
...
+        await request(app.getHttpServer())
+            .post('/api/auth/login')
+            .send({username: 'bruno', password: 'pws'})
+            .expect(401);
+    });
+
+    it('should not login due to wrong username', async () => {
+        await request(app.getHttpServer())
+            .post('/api/auth/login')
+            .send({username: 'usrnamenotavalibl', password: 'pws'})
```
Comment (bariod) [inline packages/api/test/e2e/app/app.e2e-spec.ts:42]:
```suggestion
            .send({username: 'usrnamenotavalible', password: 'pws'})
```

File: packages/api/http/REGISTER.http [hunk 4]
```diff
@@ -0,0 +1,9 @@
+POST http://localhost:3000/api/auth/register
+Content-Type: application/json
+
+{
+  "username": "brunos",
```
Comment (bariod) [inline packages/api/http/REGISTER.http:5]:
```suggestion
  "username": "bruno",
```
intentional?

**Review by bariod — CHANGES_REQUESTED** [review]

File: packages/api/src/modules/auth/auth.controller.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+@Controller('/api/auth')
+export class AuthenticationController {
+  private readonly users: UsersService;
+
+  private readonly tokens: TokensService;
+
+  /**
+   *
```
[the remark this answers is not in the document]
Comment (9j3) [reply packages/api/src/modules/auth/auth.controller.ts:23]: These are required for upcoming API documentation with swagger

File: packages/api/test/e2e/app/app.e2e-spec.ts (hunk 2 above)
Comment (9j3) [reply packages/api/test/e2e/app/app.e2e-spec.ts:4]: These are .spec files (for testing) which are ignored by ESLint. I can add them if you wan't but atm I think it doesn't matter that much :)

File: packages/api/src/modules/auth/token.service.ts [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+export interface RefreshTokenPayload {
+  jti: number;
+  sub: number
+}
+
+@Injectable()
+export class TokensService {
+  private readonly tokens: RefreshTokensRepository;
+
```
[the remark this answers is not in the document]
Comment (9j3) [reply packages/api/src/modules/auth/token.service.ts:28]: Our style guide allows either no line or one line :)

File: packages/api/test/e2e/app/app.e2e-spec.ts [hunk 7]
```diff
@@ -39,7 +39,7 @@ describe('E2E Login', () => {
     it('should not login due to wrong username', async () => {
         await request(app.getHttpServer())
             .post('/api/auth/login')
-            .send({username: 'usrnamenotavalibl', password: 'pws'})
+            .send({username: 'usrnamenotavalible', password: 'pws'})
```
Comment (bariod) [inline packages/api/test/e2e/app/app.e2e-spec.ts:42]:
```suggestion
            .send({username: 'usernamenotavalible', password: 'pws'})
```
aaaah sryyy

**Review by bariod — APPROVED** [review]

**Pull request merged**